### PR TITLE
Add all current models and add properties to existing models

### PR DESCRIPTION
**Pull request by silvant**

### CODE REVIEW

File: src/Picqer/Financials/Exact/BankAccount.php [hunk 1]
```diff
@@ -4,60 +4,53 @@
...
-        'BankDescription',
-        'BankName',
         'BICCode',
         'Created',
         'Creator',
         'CreatorFullName',
         'Description',
         'Division',
         'Format',
-        'IBAN',
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/BankAccount.php]: Do these obsolete properties not work at all anymore? Or do they still work, but only marked as obsolete?

File: src/Picqer/Financials/Exact/CashEntry.php [hunk 2]
```diff
@@ -1,67 +1,50 @@
...
+        'Status',
+        'StatusDescription'
+    ];
+
+    protected $url = 'financialtransaction/CashEntries';
 
-	use Query\Findable;
-	use Persistance\Storable;
-
-	protected $primaryKey = 'EntryID';
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/CashEntry.php]: The default primary key for an entity is 'ID'. This removes the overridden primary key for this entity.

File: src/Picqer/Financials/Exact/CashEntry.php [hunk 3]
```diff
@@ -1,67 +1,50 @@
...
-		'JournalCode',
-		'JournalDescription',
-		'Modified',
-		'OpeningBalanceFC',
-		'Status',
-		'StatusDescription'
-	];
-
-
-	public function addItem(array $array)
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/CashEntry.php]: Why is this specific function removed?

File: src/Picqer/Financials/Exact/CostsByDate.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    use Query\Findable;
+    use Query\Relatable;
+
+    /**
+     * @var string Name of the parent key for this model
+     */
+    protected $parentKey = 'checkDate';
+
+    protected $fillable = [
+        'Id',
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/CostsByDate.php]: We might need to specify a specific primary key on this one, since this is `Id` instead of `ID`.

File: src/Picqer/Financials/Exact/CostsById.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    use Query\Findable;
+    use Query\Relatable;
+
+    /**
+     * @var string Name of the parent key for this model
+     */
+    protected $parentKey = 'entryId';
+
+    protected $fillable = [
+        'Id',
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/CostsById.php]: We might need to specify a specific primary key on this one, since this is `Id` instead of `ID`.

File: src/Picqer/Financials/Exact/Currency.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+ * @property String $Description Description of the currency
+ * @property DateTime $Modified Last modified date
+ * @property Double $PricePrecision Price precision
+ */
+class Currency extends Model
+{
+    use Query\Findable;
+
+    protected $fillable = [
+        'Code',
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/Currency.php]: Need to specify this as the primary key.

File: src/Picqer/Financials/Exact/Division.php [hunk 7]
```diff
@@ -4,61 +4,36 @@
...
- * @property String $TaxReferenceNumber Local tax reference number (Germany)
- * @property String $VATNumber VAT number
- * @property String $Website Customer value, hyperlink to external website
+ * @property String $CustomerCode Owner account code of the division
+ * @property String $CustomerName Owner account name of the division
+ * @property String $Description Description
+ * @property Int64 $Hid Company number that is assigned by the customer
+ * @property Boolean $IsMainDivision True if the division is the main division
  */
 class Division extends Model
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/Division.php]: This is now reformed to SystemsDivision but was meant as HRMDivision. This will break for people using HRM Divisions with the client.

Comment (stephangroen) [inline src/Picqer/Financials/Exact/Division.php]: If you'd like to add SystemsDivisions, it should be called just that.

File: src/Picqer/Financials/Exact/GeneralJournalEntry.php [hunk 8]
```diff
@@ -1,44 +1,54 @@
...
+ * @property Int32 $Type Type: 10 = Opening balance, 90 = Other
+ * @property String $TypeDescription Description of Type
  */
 class GeneralJournalEntry extends Model
 {
-
     use Query\Findable;
     use Persistance\Storable;
 
-    protected $primaryKey = 'EntryID';
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/GeneralJournalEntry.php]: Why is this removed as EntryID is the proper primary key?

File: src/Picqer/Financials/Exact/GoodsDelivery.php [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+ * @property String $WarehouseCode Code of Warehouse
+ * @property String $WarehouseDescription Description of Warehouse
+ */
+class GoodsDelivery extends Model
+{
+    use Query\Findable;
+    use Persistance\Storable;
+
+    protected $fillable = [
+        'EntryID',
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/GoodsDelivery.php]: Primary key EntryID not specified.

File: src/Picqer/Financials/Exact/PrintedSalesInvoice.php [hunk 10]
```diff
@@ -23,12 +23,13 @@
  * @property Int32 $ReportingPeriod Reporting period
  * @property Int32 $ReportingYear Reporting year
  * @property Boolean $SendEmailToCustomer Set to True if an email containing the invoice should be sent to the invoice customer
+ * @property String $SenderEmailAddress Email address from which the email will be sent. If not specified, the company email address will be used.
  * @property Boolean $SendInvoiceToCustomerPostbox Set to True if a postbox message containing the invoice should be sent to the invoice customer
  * @property Boolean $SendOutputBasedOnAccount Set to True if the output preference should be taken from the account. It will be either Paper, Email, Digital postbox. This option overrules both SendEmailToCustomer and SendInvoiceToCustomerPostbox.
  */
 class PrintedSalesInvoice extends Model
 {
-    protected $primaryKey = 'InvoiceID';
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/PrintedSalesInvoice.php]: Correct primary key removed?

File: src/Picqer/Financials/Exact/PurchaseEntry.php [hunk 11]
```diff
@@ -34,49 +56,53 @@ class PurchaseEntry extends Model
     use Query\Findable;
     use Persistance\Storable;
 
-    protected $primaryKey = 'EntryID';
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/PurchaseEntry.php]: Correct primary key removed?

File: src/Picqer/Financials/Exact/PurchaseEntry.php [hunk 12]
```diff
@@ -34,49 +56,53 @@ class PurchaseEntry extends Model
...
+        'SupplierName',
+        'Type',
+        'TypeDescription',
+        'VATAmountDC',
         'VATAmountFC',
-        'YourRef',
+        'YourRef'
     ];
 
-    public function addItem(array $array)
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/PurchaseEntry.php]: Why is this removed?

File: src/Picqer/Financials/Exact/SalesInvoice.php [hunk 13]
```diff
@@ -57,12 +65,9 @@
  */
 class SalesInvoice extends Model
 {
-
     use Query\Findable;
     use Persistance\Storable;
 
-    protected $primaryKey = 'InvoiceID';
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/SalesInvoice.php]: PK removed :)

File: src/Picqer/Financials/Exact/SalesInvoice.php [hunk 14]
```diff
@@ -116,17 +129,4 @@ class SalesInvoice extends Model
 
     protected $url = 'salesinvoice/SalesInvoices';
 
-    /**
-     * Updates the SalesInvoiceLines collection on a SalesInvoice if it's been detected as a deferred collection.
-     * Fetches results and stores them on this object.
-     *
-     * @return mixed
-     */
-    public function getSalesInvoiceLines() {
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/SalesInvoice.php]: Why is this removed?

File: src/Picqer/Financials/Exact/StockPosition.php [hunk 15]
```diff
@@ -1,39 +1,33 @@
...
-     * @var string
-     */
     protected $url = 'read/logistics/StockPosition';
 
-    /**
-     * The primary key for the current entity.
-     *
-     * @var string
-     */
-    protected $primaryKey = 'ItemId';
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/StockPosition.php]: PK

File: src/Picqer/Financials/Exact/Transaction.php [hunk 16]
```diff
@@ -4,53 +4,70 @@
...
+ * @property Int32 $Type 3000 = Budget
  * @property String $TypeDescription The description of the transaction type
  */
 class Transaction extends Model
 {
     use Query\Findable;
-    use Persistance\Storable;
-    
-    protected $primaryKey = 'EntryID';
```
Comment (stephangroen) [inline src/Picqer/Financials/Exact/Transaction.php]: PK

File: src/Picqer/Financials/Exact/BankAccount.php (hunk 1 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/BankAccount.php]: Obsolete, you want to keep it? I propose we include/keep obsolete properties, but maybe mark then in comments. Throwing a notice might be even the nicest way.

File: src/Picqer/Financials/Exact/CashEntry.php (hunk 2 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/CashEntry.php]: Your right, I will fix the generator

File: src/Picqer/Financials/Exact/CashEntry.php (hunk 3 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/CashEntry.php]: Sorry generator, and I missed it in the merge, I will readd it

File: src/Picqer/Financials/Exact/CostsByDate.php (hunk 4 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/CostsByDate.php]: Let's do that, should be fixed when I update the generator to include primary keys

File: src/Picqer/Financials/Exact/Division.php (hunk 7 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/Division.php]: Ok

File: src/Picqer/Financials/Exact/StockPosition.php (hunk 15 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/StockPosition.php]: Hmm, this is a function and therefor doesn't have a primary key. (There can be many rows of stockPosition having the same ItemId.) What shall we do?

File: src/Picqer/Financials/Exact/CostsById.php (hunk 5 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/CostsById.php]: Done

File: src/Picqer/Financials/Exact/CostsByDate.php (hunk 4 above)
Comment (silvant) [inline src/Picqer/Financials/Exact/CostsByDate.php]: Done
